File: benchmarks/baselines/cross-language/sieve.go
package main

import "os"

func main() {
	n := 1000000
	isPrime := make([]bool, n+1)
	for i := range isPrime { isPrime[i] = true }
	isPrime[0], isPrime[1] = false, false
	for p := 2; p*p <= n; p++ {
		if isPrime[p] {
			for m := p * p; m <= n; m += p { isPrime[m] = false }
		}
	}
	count := 0
	for _, b := range isPrime { if b { count++ } }
	if count == 78498 { os.Exit(0) } else { os.Exit(1) }
}

File: benchmarks/baselines/cross-language/sieve.py
import sys
def sieve(limit):
    is_prime = [True] * (limit + 1)
    is_prime[0] = is_prime[1] = False
    p = 2
    while p * p <= limit:
        if is_prime[p]:
            for m in range(p * p, limit + 1, p):
                is_prime[m] = False
        p += 1
    return is_prime

primes = sieve(1_000_000)
count = sum(primes)
sys.exit(0 if count == 78498 else 1)
